src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1055,30 +1070,57 @@ public override Task<object> ExecuteScalarAsync(CancellationToken cancellationTo
         /// Unlike the ADO.NET method which it replaces, this method returns a Npgsql-specific
         /// DataReader.
         /// </remarks>
-        /// <returns>A DbDataReader object.</returns>
-        public new NpgsqlDataReader ExecuteReader(CommandBehavior behavior) => (NpgsqlDataReader) base.ExecuteReader(behavior);
+        /// <returns>A NpgsqlDataReader object.</returns>
+        public new NpgsqlDataReader ExecuteReader(CommandBehavior behavior)
+            => (NpgsqlDataReader) base.ExecuteReader(behavior);
 
         /// <summary>
-        /// Executes the command text against the connection.
+        /// An asynchronous version of <see cref="ExecuteReader()"/>, which executes
+        /// the <see cref="CommandText"/> against the <see cref="Connection"/>
+        /// and returns a <see cref="NpgsqlDataReader"/>.
         /// </summary>
-        /// <param name="behavior">An instance of <see cref="CommandBehavior"/>.</param>
-        /// <param name="cancellationToken">A task representing the operation.</param>
-        /// <returns></returns>
-        protected override Task<DbDataReader> ExecuteDbDataReaderAsync(CommandBehavior behavior, CancellationToken cancellationToken)
+        /// <returns>A task representing the asynchronous operation.</returns>
+        public new Task<NpgsqlDataReader> ExecuteReaderAsync()
+            => ExecuteReaderAsync(CommandBehavior.Default, CancellationToken.None);
+
+        /// <summary>
+        /// An asynchronous version of <see cref="ExecuteReader()"/>, which executes
+        /// the <see cref="CommandText"/> against the <see cref="Connection"/>
+        /// and returns a <see cref="NpgsqlDataReader"/>.
+        /// </summary>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests.</param>
+        /// <returns>A task representing the asynchronous operation.</returns>
+        public new Task<NpgsqlDataReader> ExecuteReaderAsync(CancellationToken cancellationToken)
+            => ExecuteReaderAsync(CommandBehavior.Default, cancellationToken);
+
+        /// <summary>
+        /// An asynchronous version of <see cref="ExecuteReader(CommandBehavior)"/>,
+        /// which executes the <see cref="CommandText"/> against the <see cref="Connection"/>
+        /// and returns a <see cref="NpgsqlDataReader"/>.
+        /// </summary>
+        /// <param name="behavior">One of the enumeration values that specified the command behavior.</param>
+        /// <returns>A task representing the asynchronous operation.</returns>
+        public new Task<NpgsqlDataReader> ExecuteReaderAsync(CommandBehavior behavior)
```

Comment:
Since this is a new API, add default and merge with parameter-less overload above.

Reply:
Thought that `DbCommand.ExecuteReaderAsync(CommandBehavior)` will take precedence over `NpgsqlCommand.ExecuteReaderAsync(CommandBehavior, CancellationToken)`, but the compiler understands that correctly. It hides both base methods.